src/main/webapp/app/exam/participate/summary/exam-result-summary.component.html

```diff
@@ -69,108 +49,123 @@ <h5>
-                        *ngIf="exercise.includedInOverallScore !== IncludedInOverallScore.INCLUDED_COMPLETELY"
-                        [includedInOverallScore]="exercise.includedInOverallScore"></jhi-included-in-score-badge>
-                </h5>
             </div>
-            <h6 right-body class="markdown-preview" *ngIf="exercise.problemStatement && exercise.type !== PROGRAMMING">
-                <span [innerHTML]="exercise.problemStatement! | htmlForMarkdown"></span>
-            </h6>
-            <jhi-programming-exercise-instructions
-                *ngIf="exercise.problemStatement && exercise.type === PROGRAMMING"
-                [exercise]="exercise"
-                [participation]="asProgrammingExercise(exercise).templateParticipation!"
-            ></jhi-programming-exercise-instructions>
-            <ng-container *ngIf="exercise.type !== PROGRAMMING">
+            <ng-container *ngIf="exercise.type !== PROGRAMMING" class="pt-2">
+                <div *ngIf="exerciseInfos[exercise.id!].displayExampleSolution; else studentSolution">
+                    <jhi-example-solution *ngIf="exercise.id !== undefined" [exerciseId]="exercise.id" [displayHeader]="false" />
+                </div>
+
                 <!-- Show submission of student - if no submission found -> display that student did not participate -->
-                <ng-container
-                    *ngIf="getParticipationForExercise(exercise) && getSubmissionForExercise(exercise) && getSubmissionForExercise(exercise)?.submitted; else noSubmission"
-                >
-                    <!-- Text Exercise -->
-                    <ng-container *ngIf="exercise.type === TEXT">
-                        <jhi-text-exam-summary [submission]="getSubmissionForExercise(exercise)!"></jhi-text-exam-summary>
-                    </ng-container>
-                    <!-- Modeling Exercise -->
-                    <ng-container *ngIf="exercise.type === MODELING">
-                        <jhi-modeling-exam-summary [exercise]="exercise" [submission]="getSubmissionForExercise(exercise)!"></jhi-modeling-exam-summary>
-                    </ng-container>
-                    <!-- Quiz Exercise -->
-                    <ng-container *ngIf="exercise.type === QUIZ">
-                        <jhi-quiz-exam-summary
+                <ng-template #studentSolution>
+                    <ng-container *ngIf="exerciseInfos[exercise.id!]?.participation && exerciseInfos[exercise.id!]?.submission?.submitted; else noSubmission">
+                        <ng-container *ngIf="exercise.type === TEXT">
+                            <jhi-text-exam-summary
+                                [exercise]="exercise"
+                                [submission]="exerciseInfos[exercise.id!]!.submission!"
+                                [expandProblemStatement]="expandProblemStatement"
+                                [isAfterResultsArePublished]="isAfterResultsArePublished()"
```

Comment:
Do you want the page to dynamically change as soon as the results are published? If not, please make this an attribute to avoid constantly recalling this method.

Reply:
It was an active decision to leave the checks based on time as method calls - however, for the page to update properly, more actions would be needed, so this can be set once onInit.

If we want the page to update automatically, I think we should have a dedicated look at solving this well.

Reply:
So I removed the method calls for now